### PR TITLE
Add change log & change order of args in NonceGen hash fn

### PR DESCRIPTION
**Pull request by jonasnick**

### CODE REVIEW

File: bip-musig2.mediawiki [hunk 1]
```diff
@@ -549,6 +549,15 @@ There are, however, insecure toy schemes with AGM security proofs, but those exp
 In contrast, in MuSig2, all group elements that arise in the protocol are known to the adversary and declared as group element inputs.
 A scheme very similar to MuSig2 and with two-point nonces was independently proven secure in the ROM and AGM by [https://eprint.iacr.org/2020/1245 Alper and Burdges].
 
+== Change Log ==
+
+To help implementors understand updates to this BIP, we attach a version number that resembles ''semantic versioning'' (<code>MAJOR.MINOR.PATCH</code>).
+A change in the <code>MAJOR</code> version indicates that the specification is incompatible with prior versions.
+The <code>MINOR</code> version is increased whenever the inputs or the output of an algorithm changes in a backward compatible way.
```
Comment (real-or-random) [inline bip-musig2.mediawiki]: When do we increase `PATCH`? For *every* other change, or are we allowed to rephrase things without a patch? 

For example: Do we increase `PATCH` if
 * we fix a typo?
 * improve an explanation?
 * change an algorithm without changing its I/O behavior?

What do we change if we rename an algorithm?

We don't perfect answers here. This is not actual semantic versioning but we should have some guidelines in mind.

I found this by the way: https://datatracker.ietf.org/doc/html/draft-claise-semver-02

File: bip-musig2.mediawiki [hunk 2]
```diff
@@ -549,6 +549,15 @@ There are, however, insecure toy schemes with AGM security proofs, but those exp
...
+
+* '''0.2.0''' (2022-04-19): Change order of arguments in ''NonceGen'' hash function
```
Comment (real-or-random) [inline bip-musig2.mediawiki]: Now a month has passed... (Sorry for that!) But even if we're quick, we won't always merge on the same day. Should we stick to dates here? Should we insert the dates only merging? Should we use commit ids instead? (commit ids are problematic if we PR to the official BIPs repo later but doing it for now wouldn't be too bad and then we could replace by dates later?)

**Review by real-or-random — COMMENTED** [review]
ACK on the actual change

Strong concept ACK on the changelog

File: bip-musig2.mediawiki (hunk 1 above)
Comment (jonasnick) [reply bip-musig2.mediawiki]: Good find! I had thought we increment it whenever we do something that's should get the attention of implementors. So to answer your questions: no, depends, depends. Added a sentence for that and harmonized with the ietf spec. Or should we just refer to it?

File: bip-musig2.mediawiki (hunk 2 above)
Comment (jonasnick) [reply bip-musig2.mediawiki]: Right now it's the date of the commit. This helps finding the commit in the log even without the commit id. If it's not too distant from the merge I think that's fine and simpler than requiring the date to be the same as the merge commit. Bumped the dates.

**Review by real-or-random — APPROVED** [review]
ACK aab260c37112b2010266fd8f0d3d56ea1111a2cc

File: bip-musig2.mediawiki (hunk 1 above)
Comment (real-or-random) [reply bip-musig2.mediawiki]: sounds good and pragmatic

File: bip-musig2.mediawiki (hunk 2 above)
Comment (real-or-random) [reply bip-musig2.mediawiki]: Same here, sounds good and pragmatic :)

**Pull request merged**